This fixes #52740: crash when localizing a symtab entry [1]
It was pulled to -current in 33807 which was superseded by 34213.
For 5.10.x, this was pulled in by 34295.

diff -up perl-5.10.0/sv.c.stlocal perl-5.10.0/sv.c
--- perl-5.10.0/sv.c.stlocal	2008-10-12 10:31:28.000000000 +0200
+++ perl-5.10.0/sv.c	2008-10-12 10:31:39.000000000 +0200
@@ -3546,7 +3546,7 @@ Perl_sv_setsv_flags(pTHX_ SV *dstr, regi
 	    Perl_croak(aTHX_ "Cannot copy to %s", type);
     } else if (sflags & SVf_ROK) {
 	if (isGV_with_GP(dstr) && dtype == SVt_PVGV
-	    && SvTYPE(SvRV(sstr)) == SVt_PVGV) {
+	    && SvTYPE(SvRV(sstr)) == SVt_PVGV && isGV_with_GP(SvRV(sstr))) {
 	    sstr = SvRV(sstr);
 	    if (sstr == dstr) {
 		if (GvIMPORTED(dstr) != GVf_IMPORTED
